ios/FluentUI/Vnext/AvatarGroup/AvatarGroup.swift

```diff
@@ -125,36 +125,43 @@ public struct AvatarGroup: View {
-                let stackPadding = (currentAvatarHasRing ? ringPaddingInterspace : noRingPaddingInterspace)
-
-                let xOrigin = currentAvatarHasRing ? x + ringOffset : x
-                let yOrigin = currentAvatarHasRing ? (nextAvatarHasRing ? ringOuterGap : ringOffset) :
-                    (nextAvatarHasRing ? 0 - ringOffset + tokens.ringOuterGap : 0)
-                let cutoutSize = nextAvatarHasRing ? size + ringOffset + ringOuterGap : size
-
-                AvatarView(avatar)
-                    .modifyIf(needsCutout, { view in
-                        view.mask(AvatarCutout(xOrigin: xOrigin,
-                                               yOrigin: yOrigin,
-                                               cutoutSize: cutoutSize)
-                                    .fill(style: FillStyle(eoFill: true)))
-                    })
-                    .padding(.trailing, tokens.style == .stack ? stackPadding : interspace)
-            }
-            if overflowCount > 0 {
-                createOverflow(count: overflowCount)
+        let groupHeight: CGFloat = size + (ringOffset * 2)
+
+        GeometryReader { geo in
+            HStack(spacing: 0) {
+                ForEach(0 ..< maxDisplayedAvatars, id: \.self) { index in
+                    // If the avatar is part of Stack style and is not the last avatar in the sequence, create a cutout
+                    let needsCutout = tokens.style == .stack && (overflowCount > 0 || index + 1 < maxDisplayedAvatars)
+                    let avatar = avatars[index]
+                    let currentAvatarHasRing = avatar.isRingVisible
+                    let nextAvatarHasRing = index + 1 < maxDisplayedAvatars ? avatars[index + 1].isRingVisible : false
+
+                    let ringPaddingInterspace = nextAvatarHasRing ? interspace - (ringOffset + ringOuterGap) : interspace - ringOffset
+                    let noRingPaddingInterspace = nextAvatarHasRing ? interspace - ringOuterGap : interspace
+                    let stackPadding = (currentAvatarHasRing ? ringPaddingInterspace : noRingPaddingInterspace)
+
+                    let xOrigin = currentAvatarHasRing ? x + ringOffset : x
+                    let yOrigin = currentAvatarHasRing ? (nextAvatarHasRing ? ringOuterGap : ringOffset) :
+                        (nextAvatarHasRing ? 0 - ringOffset + tokens.ringOuterGap : 0)
+                    let cutoutSize = nextAvatarHasRing ? size + ringOffset + ringOuterGap : size
+                    AvatarView(avatar)
+                        .modifyIf(needsCutout, { view in
+                            view.mask(AvatarCutout(xOrigin: xOrigin,
+                                                   yOrigin: yOrigin,
+                                                   cutoutSize: cutoutSize)
+                                        .fill(style: FillStyle(eoFill: true)))
+                        })
+                        .padding(.trailing, tokens.style == .stack ? stackPadding : interspace)
```

Comment:
So, the GeometryReader won't be able to replace the padding to position the Avatars in the View?
I thought that this would be replaced by a .frame(x: y:) modifier.